### PR TITLE
feat(autoware_launch): add controller mode options to launch file

### PR DESCRIPTION
**Pull request by brkay54**
## Description

This PR enables user to specify the lateral and longitudinal controller type. 
It should be merged after [this PR](https://github.com/autowarefoundation/autoware_launch/pull/106) merged!

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/launch/planning_simulator.launch.xml [hunk 1]
```diff
@@ -47,6 +51,10 @@
       <!-- Map -->
       <arg name="lanelet2_map_file" value="$(var lanelet2_map_file)"/>
       <arg name="pointcloud_map_file" value="$(var pointcloud_map_file)"/>
+      <!-- Control -->
+      <arg name="enable_obstacle_collision_checker" value="$(var enable_obstacle_collision_checker)"/>
+      <arg name="lateral_controller_mode" value="$(var lateral_controller_mode)"/>
+      <arg name="longitudinal_controller_mode" value="$(var longitudinal_controller_mode)"/>
```
Comment (takayuki5168) [inline autoware_launch/launch/planning_simulator.launch.xml:57]: Could you check whether the option works when launching planning_simulator.launch.xml even if we discard the change of planning_simulator.launch.xml.

Comment (brkay54) [reply autoware_launch/launch/planning_simulator.launch.xml:57]: "discard the change of planning_simulator.launch.xml": you mean that different options in autoware.launch.xml and planning_simulator.launch.xml right? Do we want to test that planning_simulator.launch.xml can overwrites the autoware.launch.xml properly?

Comment (takayuki5168) [reply autoware_launch/launch/planning_simulator.launch.xml:57]: No the different options.

This is an example of tier4/autoware_launch
[cruise_planner_type in autoware.launch.xml](https://github.com/tier4/autoware_launch/blob/e282e15d4bddcebec6401ecdc5cf525a71cfff20/autoware_launch/launch/autoware.launch.xml#L14) can be designated when launching [planning_simualtor.launch.xml](https://github.com/tier4/autoware_launch/blob/awf-latest/autoware_launch/launch/planning_simulator.launch.xml) like `ros2 launch autoware_launch planning_simulator.launch.xml cruise_planner_type:=obstacle_stop_planner` even though planning_simulator.launch.xml does not have an argument of `cruise_planner_type`.

Therefore, in the same way, in this PR, we do not have to add longitudinal_controller_mode in planning_simulator.launch.xml.

Comment (brkay54) [reply autoware_launch/launch/planning_simulator.launch.xml:57]: Oh I see, I tested both scenarios (with removing `lateral_controller_mode` from planning_simulator.launch.xml and without removing it). We can assign mode from terminal (like you said), and it worked.

Comment (takayuki5168) [reply autoware_launch/launch/planning_simulator.launch.xml:57]: So could you remove `lateral_controller_mode` from launch files other than `autoware.launch.xml`. It worked well, right?

Comment (brkay54) [reply autoware_launch/launch/planning_simulator.launch.xml:57]: Yes, I removed from planning_simulator.launch.xml and worked.

Comment (brkay54) [reply autoware_launch/launch/planning_simulator.launch.xml]: Sorry for misunderstanding, my college just warned me. I removed options from  `planning_simulator.launch.xml` and pushed new commit. Would you check it again?

**Comment by xmfcx** [comment]
@brkay54 could you rebase this PR?

@takayuki5168 could you review this again? This is urgent for the bus odd project.

**Comment by takayuki5168** [comment]
@brkay54 @xmfcx Sorry for the delay due to the vacation.
Recently, almost the same change was merged to awf/autoware_launch.

https://github.com/autowarefoundation/autoware_launch/blob/0d5e54ab5ae9aa8d7fb9f65440f412f116d98635/autoware_launch/launch/autoware.launch.xml#L126-L127

**Comment by brkay54** [comment]
@takayuki5168 I see, thank you! I am going to close the PR. However, because these parameters were not declared as global parameters in `autoware.launch.xml`, we can not change them in command line. I am going to create PR for this.

**Pull request closed without merging**